### PR TITLE
feat: support more than one currency for FXP participants

### PR DESCRIPTION
**Pull request by kalinkrustev**
The intended result is to have this in vault:
image

### CODE REVIEW

File: src/db/migrations/20240820130000_create_fxp_supported_currencies_table.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    table.string('monetaryZoneId', 3);
+    table.foreign('monetaryZoneId').references('monetaryZoneId').inTable('monetaryZone');
+    table.foreign('dfspId', 'FK_CURR_DFSP_ID').references('dfsps.id').onDelete('CASCADE').onUpdate('NO ACTION');
+    table.index('dfspId', 'FK_CURR_DFSP_ID_idx');
+    table.unique(['dfspId', 'monetaryZoneId']);
+    if (!process.env.TEST) table.engine('InnoDB');
+    if (!process.env.TEST) table.charset('utf8mb4');
+  });
+};
+
```
Comment (vijayg10) [inline src/db/migrations/20240820130000_create_fxp_supported_currencies_table.js:14]: May be the table name can be generalised as normal dfsps can also support multiple currencies.

File: src/api/swagger.yaml [hunk 2]
```diff
@@ -2776,6 +2776,11 @@ components:
         monetaryZoneId:
           type: string
           description: The monetary Zone
+        fxpCurrencies:
+          type: array
+          description: List of FX currencies
+          items:
+            type: string
```
Comment (vijayg10) [inline src/api/swagger.yaml:2783]: I think this field can be generalised like 'supportedCurrencies'. Normal DFSPs can also support multiple currencies.

Comment (kalinkrustev) [reply src/api/swagger.yaml:2783]: This is supposed to be used specifically for FX currencies when adding FXP participant. As `supportedCurrencies` are something that is used in the ALS result, this property is to distinguish from that usage case. So we can have additional property for that case.

File: src/db/migrations/20240820130000_create_fxp_supported_currencies_table.js (hunk 1 above)
Comment (kalinkrustev) [reply src/db/migrations/20240820130000_create_fxp_supported_currencies_table.js:14]: Let's have another table for normal dfsps. This table indicates that this is an FXP participant.

**Review by vijayg10 — APPROVED** [review]

**Pull request merged**